From 07732114a20ddf6e7de9b6d99733c8727587c729 Mon Sep 17 00:00:00 2001
From: Artur Wieczorek <artwik@wp.pl>
Date: Wed, 22 Apr 2015 20:51:47 +0200
Subject: Add to the test unit testing Unicode file names.

The purpose of the test is to check if file names (also with Unicode characters) are converted properly (in wxString::fn_str) to the internal representation of file names using by the file system.

diff --git a/tests/filename/filenametest.cpp b/tests/filename/filenametest.cpp
index b44a288..d330a51 100644
--- a/tests/filename/filenametest.cpp
+++ b/tests/filename/filenametest.cpp
@@ -151,6 +151,7 @@ private:
 #ifdef __WINDOWS__
         CPPUNIT_TEST( TestShortcuts );
 #endif // __WINDOWS__
+        CPPUNIT_TEST( TestUnicode );
     CPPUNIT_TEST_SUITE_END();
 
     void TestConstruction();
@@ -177,7 +178,7 @@ private:
 #ifdef __WINDOWS__
     void TestShortcuts();
 #endif // __WINDOWS__
-
+    void TestUnicode();
     DECLARE_NO_COPY_CLASS(FileNameTestCase)
 };
 
@@ -1044,3 +1045,26 @@ void FileNameTestCase::TestShortcuts()
 }
 
 #endif // __WINDOWS__
+
+void FileNameTestCase::TestUnicode()
+{
+#if wxMBFILES
+    const char *pfn;
+#else
+    const wxChar *pfn;
+#endif
+    wxString fn = wxT("wx_file_mask.txt");
+    pfn = fn.fn_str();
+    wxString msg = wxString::Format(wxT("File name: %s"),
+                                          fn.c_str());
+    const char *pUnitMsg = (const char*)msg.mb_str(wxConvUTF8);
+    CPPUNIT_ASSERT_MESSAGE( pUnitMsg, wxStrlen(pfn) > 0 );
+
+    // This file name contains UTF-16 characters
+    // ('wx_file_mask.txt' in Russian).
+    fn = wxT("wx_\x043C\x0430\x0441\x043A\x0430_\x0444\x0430\x0438\x043F\x0430.txt");
+    pfn = fn.fn_str();
+    msg = wxString::Format(wxT("File name: %s"), fn.c_str());
+    pUnitMsg = (const char*)msg.mb_str(wxConvUTF8);
+    CPPUNIT_ASSERT_MESSAGE( pUnitMsg, wxStrlen(pfn) > 0 );
+}
-- 
1.9.5.msysgit.1

